tests/unit/helpers.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""helpers for the unit test."""
+
+import contextlib
+import typing
+import unittest.mock
+
+import ops
+from ops.model import Container
+from ops.testing import Harness
+
+from charm import DiscourseCharm
+
+DATABASE_NAME = "discourse"
+
+
+@contextlib.contextmanager
+def patch_exec(fail: bool = False) -> typing.Generator[unittest.mock.Mock, None, None]:
+    """Patch the ops.model.Container.exec method.
+
+    When fail argument is true, the execution will fail.
+
+    Yields:
+        Mock for the exec method.
+    """
+    exec_process_mock = unittest.mock.MagicMock()
+    if not fail:
+        exec_process_mock.wait_output = unittest.mock.MagicMock(return_value=("", ""))
+    else:
+        exec_process_mock.wait_output = unittest.mock.Mock()
+        exec_process_mock.wait_output.side_effect = ops.pebble.ExecError([], 1, "", "")
+    exec_function_mock = unittest.mock.MagicMock(return_value=exec_process_mock)
+    with unittest.mock.patch.multiple(ops.model.Container, exec=exec_function_mock):
+        yield exec_function_mock
+
+
+def start_harness(
```

Comment:
What an interesting approach! :clap: 

I tried to use fixtures to solve this problem of setting a harness with specific conditions for multiple tests (see discussion [here](https://github.com/canonical/synapse-operator/pull/19#discussion_r1302486670) in Synapse) but this, at least for me, it's clearer since you are using named parameters.